### PR TITLE
return exception and report to sentry

### PR DESCRIPTION
**Pull request by ahmardi**

### CODE REVIEW

**Review by Skullsneeze — APPROVED** [review]

**Review by sameh-amwal — COMMENTED** [review]
please add throw exception also in dataValidation or remove the try exception block

File: Model/Data/OrderUpdate.php [hunk 1]
```diff
@@ -142,7 +146,7 @@ public function update($order, $trigger, $sendAdminEmail = true)
             return $status == 'success'? $amwalOrderData : false;
         } catch (\Exception $e) {
             $this->sentryExceptionReport->report($e->getMessage());
-            return false;
+            throw new \Exception($e->getMessage());
```
Comment (sameh-amwal) [inline Model/Data/OrderUpdate.php:149]:
```suggestion
            throw $e;
```

**Review by sameh-amwal — APPROVED** [review]

File: Model/Checkout/PayOrder.php [hunk 2]
```diff
@@ -74,7 +77,8 @@ public function __construct(
         ErrorReporter $errorReporter,
         Config $config,
         LoggerInterface $logger,
-        OrderUpdate $orderUpdate
+        OrderUpdate $orderUpdate,
+        sentryExceptionReport $sentryExceptionReport
```
Comment (Skullsneeze) [inline Model/Checkout/PayOrder.php:81]:
```suggestion
        SentryExceptionReport $sentryExceptionReport
```

File: Model/Settings.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+     * Retrieves settings.
+     *
+     * @return array An associative array of settings.
+     */
+    public function getSettings() : array
+    {
+        $settings = [
+            'amwal_payment' => $this->config->isActive(),
+            'amwal_payment_title' => $this->config->getTitle(),
+            'is_merchant_valida' => $this->config->isMerchantValid(),
```
Comment (Skullsneeze) [inline Model/Settings.php:31]:
```suggestion
            'is_merchant_valid' => $this->config->isMerchantValid(),
```

**Review by Skullsneeze — CHANGES_REQUESTED** [review]

**Review by Skullsneeze — APPROVED** [review]

**Review by Skullsneeze — APPROVED** [review]

File: Model/Data/OrderUpdate.php [hunk 4]
```diff
@@ -242,44 +245,23 @@ private function getOrderUrl(OrderInterface $order): string
...
-                            $orderMethod,
-                            $amwalMethod,
-                            $orderValue,
-                            $amwalValue
-                        )
-                    );
-                    $this->sendAdminEmail($order, 'Order (%s) needs Attention', $this->dataValidationMessage($order->getIncrementId(), $orderMethod, $amwalMethod, $orderValue, $amwalValue));
-                    throw new \Exception(sprintf('Order (%s) %s does not match Amwal Order %s (%s != %s)', $order->getIncrementId(), $orderMethod, $amwalMethod, $orderValue, $amwalValue));
-                }
+        if ($order->getOrderCurrencyCode() != $amwalOrderData->getMerchantCountryCode()) {
```
Comment (sameh-amwal) [inline Model/Data/OrderUpdate.php:248]: wouldn't be this comparing currency code (SAR) to country code (SA)?

**Pull request merged**